src/test/groovy/pl/tomaszdziurko/jvm_bloggers/rest/newsletter_issue/NewsletterIssueControllerIntegrationSpec.groovy

```diff
@@ -0,0 +1,114 @@
+package pl.tomaszdziurko.jvm_bloggers.rest.newsletter_issue
+
+import org.hamcrest.Matchers
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.SpringApplicationContextLoader
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.web.WebAppConfiguration
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+import org.springframework.test.web.servlet.setup.ConfigurableMockMvcBuilder
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+import org.springframework.web.context.WebApplicationContext
+import pl.tomaszdziurko.jvm_bloggers.JvmBloggersApplication
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPost
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPostRepository
+import pl.tomaszdziurko.jvm_bloggers.blogs.domain.Blog
+import pl.tomaszdziurko.jvm_bloggers.blogs.domain.BlogRepository
+import pl.tomaszdziurko.jvm_bloggers.newsletter_issues.domain.NewsletterIssue
+import pl.tomaszdziurko.jvm_bloggers.newsletter_issues.domain.NewsletterIssueRepository
+import spock.lang.Specification
+
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+import static pl.tomaszdziurko.jvm_bloggers.blogs.domain.BlogType.PERSONAL
+import static pl.tomaszdziurko.jvm_bloggers.rest.ContentTypes.JVM_BLOGGERS_V1
+
+@ContextConfiguration(classes = [JvmBloggersApplication], loader = SpringApplicationContextLoader)
+@WebAppConfiguration
+@ActiveProfiles("test")
+class NewsletterIssueControllerIntegrationSpec extends Specification {
```

Comment:
You can omit all the annotations by extending `SpringContextAwareSpecification` instead.